### ISSUE TITLE
some problems about using models trained with nn4_small2_v1

### ISSUE THREAD

**Issue by stringmoon** [davidsandberg/facenet#180]
Hi! For some reason,I tried to reduce the time loading the model.
I chose nn4_small2_v1 network for training and got checkpoint and meta files,but when using these for compare.py I got some errors like:

Traceback (most recent call last):
  File "compare.py", line 129, in <module>
    main(parse_arguments(sys.argv[1:]))
  File "compare.py", line 59, in main
    emb = sess.run(embeddings, feed_dict=feed_dict)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 767, in run
    run_metadata_ptr)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 965, in _run
    feed_dict_string, options, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1015, in _do_run
    target_list, options, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1035, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.FailedPreconditionError: Attempting to use uninitialized value conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage
	 [[Node: conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage/read = Identity[T=DT_FLOAT, _class=["loc:@conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage"], _device="/job:localhost/replica:0/task:0/gpu:0"](conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage)]]
	 [[Node: embeddings/_539 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_3825_embeddings", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Caused by op 'conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage/read', defined at:
  File "compare.py", line 129, in <module>
    main(parse_arguments(sys.argv[1:]))
  File "compare.py", line 50, in main
    facenet.load_model(args.model_dir, meta_file, ckpt_file)
  File "/home/fd/facenet-master(2)/src/facenet.py", line 368, in load_model
    saver = tf.train.import_meta_graph(os.path.join(model_dir_exp, meta_file))
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/training/saver.py", line 1583, in import_meta_graph
    **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/meta_graph.py", line 499, in import_scoped_meta_graph
    producer_op_list=producer_op_list)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/importer.py", line 288, in import_graph_def
    op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 2334, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1226, in __init__
    self._traceback = _extract_stack()

FailedPreconditionError (see above for traceback): Attempting to use uninitialized value conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage
	 [[Node: conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage/read = Identity[T=DT_FLOAT, _class=["loc:@conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage"], _device="/job:localhost/replica:0/task:0/gpu:0"](conv1_7x7/batch_norm/conv1_7x7/batch_norm/moments/moments_1/mean/ExponentialMovingAverage)]]
	 [[Node: embeddings/_539 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_3825_embeddings", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Is there any wrong when training?

Thanks!

**Comment by sara-dahl** [comment]
Hi @stringmoon,

Can you please share your trained model files? 
May I also ask what databases you used for its training and how long did it take for you?

@davidsandberg,
I'll also appreciate if you give an idea of how long your trainings usually take and what are the specs of the computer you use. 

Thanks,
Sara

**Comment by stringmoon** [comment]
Hi @sara-dahl ,thanks for replying.
So I just train few steps for test using lfw databases.I wonder if my training got somethings wrong.If the model files can work,I will continue my training using other larger databases.

**Comment by jasminezz** [comment]
I have the same error. Change the code
`        # Create a saver`
`        # saver = tf.train.Saver(tf.trainable_variables(), max_to_keep=3)`
to
`        saver = tf.train.Saver(tf.global_variables(), max_to_keep=3)`
will fix this issue.

**Comment by stringmoon** [comment]
Hi @jasminezz ,thanks for replying.It seems work now,so fix it.

**Comment by avasbr** [comment]
So I got this error as well and changing `tf.trainable_variables()` to `tf.global_variables()` does do the trick, but it also makes the .ckpt files quite a bit larger when training (15mb --> 117mb). Why can we not use tf.trainable_variables() in this case? I don't understand what needs to be initialized if we're loading in pre-computed weights/biases, etc, and why `tf.global_variables()` addresses that issue - I've used the saver object with `tf.trainable_variables()` in a simple convnet for MNIST and it worked just fine, but i'm lost as to why it doesn't work here.

I've also tried calling `sess.run(tf.global_variables_initializer())` prior to evaluating the tensor and it didn't seem to do anything
